3rdparty/libspng/CMakeLists.txt

```diff
@@ -0,0 +1,47 @@
+# ----------------------------------------------------------------------------
+#  CMake file for libspng. See root CMakeLists.txt
+#
+# ----------------------------------------------------------------------------
+
+project(${SPNG_LIBRARY})
+
+set(CURR_INCLUDE_DIR "${CMAKE_CURRENT_LIST_DIR}")
+set_property(GLOBAL PROPERTY SPNG_INCLUDE_DIR ${CURR_INCLUDE_DIR})
+ocv_include_directories(${CURR_INCLUDE_DIR}, ${ZLIB_INCLUDE_DIRS})
```

Comment:
> ZLIB_INCLUDE_DIRS

There is dependency on ZLIB, which could be disabled.
Need to properly handle case if zlib is disabled.

Reply:
fixed

Reply:
Is not this handled in FindLibsGrfmt CMakeList already?

Reply:
ZLIB is always "required" in this file.

Reply:
Can you please provide more information about how to handle it? I am not that familiar with CMake and dependency handling. @alalek